extensions/amp-lightbox-gallery/1.0/component.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {Lightbox} from './../../amp-lightbox/1.0/lightbox';
+import {LightboxGalleryContext} from './context';
+import {sequentialIdGenerator} from '../../../src/utils/id-generator';
+import {
+  useContext,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from '../../../src/preact';
+
+const generateLightboxItemKey = sequentialIdGenerator();
+
+/**
+ * @param {!LightboxGalleryDef.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function LightboxGallery({children}) {
+  const lightboxRef = useRef(null);
+  const lightboxElements = useRef([]);
+  const register = (key, render) => {
+    lightboxElements.current[key] = render();
+  };
+  const deregister = (key) => {
+    delete lightboxElements.current[key];
+  };
+  const context = {
+    deregister,
+    register,
+    open: () => lightboxRef.current.open(),
+  };
+  return (
+    <>
+      <Lightbox ref={lightboxRef} scrollable>
+        {/* TODO: This needs an actual close button UI */}
```

Comment:
I thought we already had close button inside the lightbox?

Reply:
We have a SR-only close button inside the lightbox currently, but I have been meaning to refactor that close button such that it is a default value for the `closeButton` prop/`slot="close-button"` feature that should be addressed separately.

For now, the `LightboxGallery` should give its own close button even if using `Lightbox` under the hood, and that close button should eventually be passed as a prop instead of a child once the above is implemented. This is because the `LightboxGallery` UI has an `X` SVG icon for a close button, so slightly different end user experiences.

Reply:
Got it.